### PR TITLE
[ON HOLD] *: update kube vendor to latest master

### PR DESCRIPTION
**Pull request by runcom**
Close https://github.com/kubernetes-incubator/cri-o/issues/509

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
@mrunalp @sameo PTAL

**Comment by runcom** [comment]
@sameo CRI has removed netns path exposure from sandbox status, therefore network tests are now failing.

**Comment by sameo** [comment]
@runcom I'll try to fix those in the next 2 days. I'm travelling this week, I'll keep you posted.

**Comment by runcom** [comment]
@sameo no worry, I'll try to replace calls to netns exec with a simple container exec followed by an ip (or ifconfig) command to get container IP and verify tests.

**Comment by runcom** [comment]
@sameo I fixed them, this should be all green now

File: test/network.bats [hunk 1]
```diff
@@ -8,8 +8,12 @@ load helpers
 	echo "$output"
 	[ "$status" -eq 0 ]
 	pod_id="$output"
+	run ocic ctr create --config "$TESTDATA"/container_redis.json --pod "$pod_id"
+	echo "$output"
+	[ "$status" -eq 0 ]
+	ctr_id="$output"
 
-	check_pod_cidr $pod_id
+	check_pod_cidr $ctr_id
```
Comment (sameo) [inline test/network.bats]: you need a `cleanup_ctrs` call here.

File: test/network.bats [hunk 2]
```diff
@@ -21,8 +25,12 @@ load helpers
 	echo "$output"
 	[ "$status" -eq 0 ]
 	pod_id="$output"
+	run ocic ctr create --config "$TESTDATA"/container_redis.json --pod "$pod_id"
+	echo "$output"
+	[ "$status" -eq 0 ]
+	ctr_id="$output"
 
-	ping_pod $pod_id
+	ping_pod $ctr_id
```
Comment (sameo) [inline test/network.bats]: ditto

File: test/network.bats [hunk 3]
```diff
@@ -34,18 +42,26 @@ load helpers
...
+	[ "$status" -eq 0 ]
+	ctr2_id="$output"
 
-	ping_pod_from_pod $pod1_id $pod2_id
+	ping_pod_from_pod $ctr1_id $ctr2_id
 	[ "$status" -eq 0 ]
 
-	ping_pod_from_pod $pod2_id $pod1_id
+	ping_pod_from_pod $ctr2_id $ctr1_id
 	[ "$status" -eq 0 ]
```
Comment (sameo) [inline test/network.bats]: ditto

**Comment by sameo** [comment]
@runcom besides the missing ctrs cleanups, LGTM.

File: test/network.bats (hunk 1 above)
Comment (runcom) [inline test/network.bats]: fixed

File: test/network.bats (hunk 2 above)
Comment (runcom) [inline test/network.bats]: fixed

File: test/network.bats (hunk 3 above)
Comment (runcom) [inline test/network.bats]: fixed

**Comment by sameo** [comment]
LGTM

File: cmd/ocic/container.go [hunk 4]
```diff
@@ -10,10 +10,10 @@ import (
 
 	"github.com/urfave/cli"
 	"golang.org/x/net/context"
+	remocommandconsts "k8s.io/apimachinery/pkg/util/remotecommand"
```
Comment (TomSweeneyRedHat) [inline cmd/ocic/container.go]: Nit, and don't hold up the submit for this, but could you go back and change this variable name to "remotecommandconsts"?  (Add 'te' after 'remo')

Comment (runcom) [inline cmd/ocic/container.go]: it's copy-pasted from kube, let's follow up on this so I don't have to rebase #511 again :)

Comment (TomSweeneyRedHat) [inline cmd/ocic/container.go]: That works for me.

**Comment by runcom** [comment]
retest

**Comment by runcom** [comment]
all, this is rebased but I'd like to get #499 in firstly

**Comment by runcom** [comment]
retest

**Comment by mrunalp** [comment]
Needs rebase now. Let us get this one in. I want to get in one more update after this to enable sharing pid namespaces.

**Comment by runcom** [comment]
rebased

**Comment by runcom** [comment]
need to investigate why node-e2e aren't working with latest kube. Still testing for #533 so @mrunalp if you could run node-e2e with this PR we may move this forward sooner.

**Comment by mrunalp** [comment]
Yeah I will run them locally.

**Comment by runcom** [comment]
> Yeah I will run them locally.

note that it needs you to have go1.8.1+

**Comment by runcom** [comment]
so tests didn't run because of https://github.com/kubernetes/kubernetes/pull/45707 which made kubelet fails to start in tests :) I'm upgrading kube again to a fixed version and node-e2e should be good to go

**Comment by runcom** [comment]
...retesting this in CI now

**Comment by runcom** [comment]
2 node-e2e failing tests are related to containers lifecycle hooks, I can reproduce the failure and looking into it now.

**Comment by runcom** [comment]
fixed failures (caused by this change in master https://github.com/kubernetes/kubernetes/pull/43395 which changed exec(s) to use `sh -c` conflicting with conmon flags parsing :smile:) I'll push a fix asap.

**Comment by runcom** [comment]
Fixed, pushed, now re-running node-e2e

**Comment by runcom** [comment]
Alright, I figured I need to make another PR for the execsync fix as it's a general issue not only related to kube bump, hold on this PR for now.

**Comment by runcom** [comment]
This is based on top of #537 so merge that and #536 before re-triggering CI here :)

**Comment by runcom** [comment]
This still requires #537 , waiting for it

**Comment by runcom** [comment]
@mrunalp oom test in Fedora fails again, could you take a look? https://aos-ci.s3.amazonaws.com/kubernetes-incubator/cri-o/crio-integration-tests-prs/355/fullresults.txt

**Comment by runcom** [comment]
@mrunalp it seems to have passed now, but that means it's definitely a flaky test. This PR should  be good to go then (we'll fix the oom test flaky in a follow up) PTAL

**Comment by runcom** [comment]
all green :green_heart:

**Comment by runcom** [comment]
@mrunalp could you please take a look at this? it's becoming a pain to keep rebasing on latest kube master :(

**Comment by runcom** [comment]
needs rebase and fix

**Comment by mrunalp** [comment]
We can get this in soon. I just want us to make sure that we manually test port forward, exec, attach etc. on this once.

**Comment by runcom** [comment]
@mrunalp I'll run kube e2e on this branch before merging so we know port forward, exec, attach etc works fine, there are new CRI routes also in kube master, and we're behind right now

**Comment by runcom** [comment]
oom test flaky, node-e2e is good here, need just rebase

File: test/helpers.bash [hunk 5]
```diff
@@ -126,11 +127,17 @@ function crio() {
...
 
-# Run crioctl using the binary specified by $OCIC_BINARY.
+# DEPRECATED
+OCIC_BINARY=${OCIC_BINARY:-${CRIO_ROOT}/cri-o/crioctl}
 function crioctl() {
 	"$OCIC_BINARY" --connect "$CRIO_SOCKET" "$@"
 }
 
+# Run crictl using the binary specified by $CRICTL_BINARY.
+function crictl() {
```
Comment (sameo) [inline test/helpers.bash]: Just as a note: We probably want the crictl integration to be part of its own commit.

Comment (runcom) [inline test/helpers.bash]: yep, will take care of it later this week probably.

Comment (runcom) [inline test/helpers.bash]: fixed now instead :)

**Comment by runcom** [comment]
we need #564 before getting this in.

File: contrib/test/crio-integration-playbook.yaml [hunk 6]
```diff
@@ -193,6 +199,14 @@
       chdir: /root/src/github.com/opencontainers/runc
     async: 600
     poll: 10
+  - name: make crictl
+    shell: |
+      go install github.com/kubernetes-incubator/cri-tools/cmd/crictl && \
```
Comment (runcom) [inline contrib/test/crio-integration-playbook.yaml]: this is breaking - we need to stick version here :)

**Comment by runcom** [comment]
I'll rework this to update master to `1.7.x` now, I'll reset `kube-1.6.x` branch to `v1.0.0-alpha.0` also -- every 1.6.x work will go there from now on

**Comment by rhatdan** [comment]
@runcom Is this still being worked?

**Comment by runcom** [comment]
Yes, kind of, it's next on my to-do list after the registry patch

**Comment by runcom** [comment]
Closing in favor of #732
